source/controllers/SubscetionController.ts

```diff
@@ -11,12 +13,168 @@ subsection.read = async (req, res) => {
+	return valid;
+};
+
+subsection.create = [
+	async (req, res): Promise<void> => {
+		if (ValidsubsectionCreateForm(req.body)) {
+			try {
+				if (isLoggedOn(res.locals.user)) {
+					if (isAdmin(res.locals.user)) {
+						await Subsection.create({
+							name: req.body.subsection_name,
+							sectionId: Number(req.body.section_id)
+						});
+					} else {
+						res.sendStatus(404);
+					}
+				} else {
+					res.sendStatus(404);
+				}
+				res.redirect(`/subsection/${req.body.section_id}`);
+			} catch (error) {
+				res.status(500);
+				res.send(error);
+			};
+		} else {
+			logger.info('Body did not match', req.body);
+			res.sendStatus(400);
+		}
+	}
+];
+
+subsection.update = subsection.handler(
+	ValidsubsectionUpdateForm,
+	async (req, res): Promise<void> => {
+		if (isLoggedOn(res.locals.user)) {
+			if (isAdmin(res.locals.user)) {
+				try {
+					const subsectioninstance = await Subsection.findByPk(req.body.subsection_id);
+					if (subsectioninstance !== null) {
+						subsectioninstance.name = req.body.subsection_name;
+						await subsectioninstance.save();
+						res.redirect(`/subsection/${req.body.section_id}`);
+					} else {
+						res.sendStatus(404);
+					}
+				} catch (error) {
+					res.send(error);
+				};
+			} else {
+				res.sendStatus(404);
+			}
+		} else {
+			res.sendStatus(404);
+		}
+	}
+);
+
+subsection.delete = async (req, res) => {
+	const subsectionId = req.body.subsec_id;
+	if (subsectionId !== null && subsectionId !== undefined && isLoggedOn(res.locals.user)) {
+		if (isAdmin(res.locals.user)) {
+			const subsectionrow = await Subsection.findByPk(subsectionId);
+			if (subsectionrow !== undefined && subsectionrow !== null) {
+				const section = subsectionrow.sectionId;
+				if (section !== undefined) {
+					await subsectionrow.destroy();
+					res.redirect(`/subsection/${section}`);
+				} else {
+					res.sendStatus(404);
+				}
+			} else {
+				res.sendStatus(404);
+			}
+		} else {
+			res.sendStatus(404);
+		}
+	} else {
+		res.sendStatus(404);
+	}
+};
+
+subsection.override('delete', '/subsection/delete');
+subsection.override('update', '/subsection/update');
+
+subsection.interface('/edit', async (req, res) => {
+	const subsecId = String(req.query.subsectionId);
+	const sectionId = String(req.params.sectionId);
+	if (isLoggedOn(res.locals.user)) {
+		if (isAdmin(res.locals.user)) {
+			const subsection = await Subsection.findByPk(subsecId);
+			const section = await Section.findByPk(sectionId);
+			if (subsection !== null && section !== null) {
+				res.render('pages/admin/subsection_edit.ejs', { sectionname: section.name, secId: section.id, oldsubsectionname: subsection.name, constants: headerConstants, userstatus_name: res.locals.user.username, userstatus_page: `/user/profile/${res.locals.user.id}`, useraccess: res.locals.user.access });
```

Comment:
ok what the fuck ir `useraccess`? idk anymore

Reply:
useraccess is from the database if you are logged in (admin and whatever placeholders are for other users), currently i have multiple placeholders for the useraccess variable so it isnt null when you arent logged in - will change it to NotLoggedIn everywhere

Reply:
May I suggest modifying the user model where you constrain the column decorator of the access property to be a string type, and modify the typescript type from `string` to `'admin' | 'client'` or instead of `'client'`, you can place whatever you decide the non-admin name. Or if we want to be funny, we could make a different `UserAccess` model, and convert the `access` property into a foreign key to this new model. That would be quite simple to make I believe and make different user access levels simpler to scale and more organised. Slap a few `bit` columns on the different access roles that indicate weather you have access to a specific feature (deleting listings, deleting images, banning etc.) Not saying we have to implement all of those but they're nice to have just in case we have the time to scale.

Reply:
The table itself is private-ish so you can make the PK an auto incrementing int column